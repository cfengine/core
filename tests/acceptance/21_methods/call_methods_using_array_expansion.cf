#######################################################
#
# Redmine 6369 - make sure that arrays don't accumulate
#
#######################################################

body common control
{
      inputs => { "../default.cf.sub" };
      bundlesequence  => { default("$(this.promise_filename)") };
      version => "1.0";
}

bundle agent init
{
  files:
      "$(G.testdir)/reports.txt"
        delete => tidy;
}
bundle agent test
{
  meta:

      "description" -> { "CFE-636" }
        string => "Check that methods can be called by expanding classic
                   arrays";

  vars:

      "b1" string => "b1";
      "b2" string => "b2";
      "b[b2]" string => "b2";

  methods:

      "${b1}"
        usebundle => ${b1};
      "${b[b2]}"
        usebundle => ${b[b2]}("param");
      "${b[b2]}"
        usebundle => ${b[${b2}]}("param2");
}

bundle agent check
{
  methods:

      "Pass/Fail"
        usebundle => dcs_check_diff( "$(G.testdir)/reports.txt",
                                     "$(this.promise_filename).expected.txt",
                                     $(this.promise_filename));
}

bundle agent b1{
  reports:
    cfengine::
      "Bundle b1"
        report_to_file => "$(G.testdir)/reports.txt";
}

bundle agent b2(ref){
  reports:
    cfengine::
      "Bundle b2 ${ref}"
        report_to_file => "$(G.testdir)/reports.txt";
}
