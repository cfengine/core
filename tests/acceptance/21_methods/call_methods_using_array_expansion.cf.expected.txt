Bundle b1
Bundle b2 param
Bundle b2 param2
